x-pack/plugins/apm/server/lib/helpers/create_or_update_index.ts

```diff
@@ -34,27 +34,36 @@ export async function createOrUpdateIndex({
   logger: Logger;
 }) {
   try {
-    const { callAsInternalUser } = esClient;
-    const indexExists = await callAsInternalUser('indices.exists', { index });
-    const result = indexExists
-      ? await updateExistingIndex({
-          index,
-          callAsInternalUser,
-          mappings
-        })
-      : await createNewIndex({
-          index,
-          callAsInternalUser,
-          mappings
-        });
+    /* Some times Kibana starts before ES is ready. When it happens an error is thrown while creating an index.
+     * To make sure create index is called again when ES is ready, it keeps trying for more 10 times.
```

Comment:
suggestion:

In some cases we could be trying to create an index before ES is ready. When this happens, we retry creating the index with exponential backoff. We use `retry`'s default formula, meaning that the first retry happens after 2s, the 5th after 32s, and the final attempt after around 17m. If the final attempt fails, the error is logged to the console.

See https://github.com/sindresorhus/p-retry and https://github.com/tim-kos/node-retry.